### ISSUE TITLE
"Invalid class method type" calling classmethod from instance on NamedTuple

### ISSUE THREAD

**Issue by Daenyth** [python/mypy#3791]
mypy==0.521

foo.py:

```python
from typing import *

class Foo(NamedTuple):
    x: int

    @classmethod
    def bar(cls) -> str:
        return cls.__name__

f = Foo(1)
y: str = f.bar() # Line 11 here
```

> foo.py:11: error: Invalid class method type

This is specific to NamedTuples. The following class doesn't trigger the same error:
```python
class Foo:
    def __init__(self, x: int) -> None:
        self.x = x

    @classmethod
    def bar(cls) -> str:
        return cls.__name__
```

**Comment by ilevkivskyi** [comment]
This is quite low priority, named tuples are intended to be "minimalistic", for all other things now-discussed dataclasses should be used.

cc @JelleZijlstra

**Comment by Daenyth** [comment]
Dataclasses would definitely solve my use case if they were in the version of python I'm using.

**Comment by ilevkivskyi** [comment]
FWIW this is fixed in master. I can reproduce the error and revealed types are correct.
